docs/root/operations/performance.rst

```diff
@@ -0,0 +1,37 @@
+.. _operations_performance:
+
+Performance
+===========
+
+Envoy is architected to optimize scalability and resource utilization by running an event loop on a
+:ref:`small number of threads <arch_overview_threading>`. The "main" thread is responsible for
+control plane processing, and each "worker" thread handles a portion of the data plane processing.
+Envoy exposes two statistics to monitor performance of the event loops on all these threads.
+
+* **Loop duration:** Some amount of processing is done on each iteration of the event loop. This
+  amount will naturally vary with changes in load. However, if one or more threads have an unusually
+  long-tailed loop duration, it may indicate a performance issue. For example, work might not be
+  distributed fairly across the worker threads, or there may be a long blocking operation in an
+  extension that's impeding progress.
+
+* **Poll delay:** On each iteration of the event loop, the event dispatcher polls for I/O events
+  and "wakes up" either when some I/O events are ready to be processed or when a timeout fires,
+  whichever occurs first. In the case of a timeout, we can measure the difference between the expected
+  wakeup time and the actual wakeup time after polling; this difference is called the "poll delay."
+  It's normal to see some small poll delay, usually equal to the kernel scheduler's "time slice" or
+  "quantum"---this depends on the specific operating system on which Envoy is running---but if this
+  number elevates substantially above its normal observed baseline, it likely indicates kernel
+  scheduler delays.
```

Comment:
This is fantastic documentation, super clear. I wish I had this on hand when I first heard about these metrics :)

Reply:
+100

Reply:
"dansplaining."